actors/evm/src/interpreter/instructions/memory.rs

```diff
@@ -53,6 +53,51 @@ pub fn get_memory_region(
     }))
 }
 
+#[inline]
+pub fn mcopy(
+    state: &mut ExecutionState,
+    _: &System<impl Runtime>,
+    dest_index: U256,
+    src_index: U256,
+    size: U256,
+) -> Result<(), ActorError> {
+    // Copy memory from src_index to dest_index.
+    // Handles overlapping slices as if using an intermediate buffer, ensuring correct copying.
+    // Expands memory if src_index + size or dest_index + size exceeds current bounds.
+    // Returns an error if memory regions are invalid or cannot be allocated.
+
+    if size > 0 {
+        copy_within_memory(&mut state.memory, dest_index, src_index, size)
+    } else {
+        Ok(())
+    }
+}
+
+pub fn copy_within_memory(
+    memory: &mut Memory,
+    dest_index: U256,
+    src_index: U256,
+    size: U256,
+) -> Result<(), ActorError> {
+    // Expand memory to accommodate requested src_index + size
+    let _region = get_memory_region(memory, src_index, size)?.expect("empty region");
+
+    // Expand memory to match dest_index + size
+    let _destination_region = get_memory_region(memory, dest_index, size)?.expect("empty region");
+
+    let src_start = src_index.low_u64() as usize;
+    let src_end = src_start + size.low_u64() as usize;
+    let dest_start = dest_index.low_u64() as usize;
```

Comment:
Any reason not to use the returned regions?

Reply:
The types here are U256 but [copy_within](https://doc.rust-lang.org/stable/core/primitive.slice.html#method.copy_within) requires usize. 

Lmk if I am missing anything either with types or with a feature that the MemoryRegion offers that I am not using.

Reply:
I mean `_region` and `_destination_region`. They use usize internally.

Reply:
Thanks! I needed that pointed out to me - 101f6e23